chain/jsonrpc/src/lib.rs

```diff
@@ -262,6 +265,18 @@ impl JsonRpcHandler {
                 match tx_status_result {
                     Ok(Ok(Some(outcome))) => break Ok(outcome),
                     Ok(Ok(None)) => {}
+                    Ok(Err(TxStatusError::MissingTransaction(_))) => {
+                        if let TransactionInfo::Transaction(tx) = &tx_info {
+                            if let Ok(NetworkClientResponses::InvalidTx(e)) =
+                                self.send_tx(tx.clone(), true).await
+                            {
+                                break Err(TxStatusError::InvalidTx(e));
+                            }
+                        }
+                        if let Ok(Err(e)) = tx_status_result {
```

Comment:
We are in the `match` arm that corresponds to `tx_status_result = Ok(Err(TxStatusError::MissingTransaction(_)))`. `tx_status_result` seems to be a local variable that doesn't change within the arm, so I'd expect this condition to be always `true`?